supermarq-benchmarks/supermarq/qcvv/base_experiment.py

```diff
@@ -16,18 +16,47 @@
 
 import functools
 import math
+import pathlib
 import warnings
 from abc import ABC, abstractmethod
 from collections.abc import Callable, Iterable, Sequence
 from dataclasses import dataclass
-from typing import Any, Generic, TypeVar
+from typing import TYPE_CHECKING, Any, Generic, TypeVar
 
 import cirq
 import cirq_superstaq as css
 import numpy as np
 import pandas as pd
 from tqdm.auto import tqdm
 
+import supermarq
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+
+def qcvv_resolver(cirq_type: str) -> type[Any] | None:
+    """Resolves string's referencing classes in the QCVV library. Used by `cirq.read_json()`
+    to deserialize.
+
+    Args:
+        cirq_type: The type being resolved
+
+    Returns:
+        The corresponding type object (if found) else None
+
+    Raises:
+        ValueError: If the provided type is not resolvable
+    """
+    prefix = "supermarq.qcvv."
+    if cirq_type.startswith(prefix):
+        name = cirq_type[len(prefix) :]
+        if name not in supermarq.qcvv.__all__:
+            raise ValueError(f"{name} is not resolvable from the QCVV library")
+        if hasattr(supermarq.qcvv, name):
+            return getattr(supermarq.qcvv, name)
```

Comment:
no need to throw an error here i think - we should just let it fall through to return `None` to fall back on the standard "could not resolve ..." error (or any other custom resolver the user might have added). in which case logic can be simplified a bit:
```suggestion
        if name in supermarq.qcvv.__all__:
            return getattr(supermarq.qcvv, name, None)
```

Reply:
Yeah nice, thanks!